src/test/java/seedu/address/logic/commands/SetThresholdCommandTest.java

```diff
@@ -0,0 +1,117 @@
+
+        String expectedMessage = String.format(SetThresholdCommand.MESSAGE_SUCCESS, VALID_THRESHOLD.goodQuantity,
+                good.getGoodName().fullGoodName);
+
+        Model expectedModel = new ModelManager(getTypicalAddressBook(), new Inventory(model.getInventory()),
+                getTypicalTransactionHistory(), new UserPrefs());
+        expectedModel.setGood(good, editedGood);
+        assertCommandSuccess(setThresholdCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_filteredList_success() {
+        showGoodAtIndex(model, INDEX_FIRST_SUPPLIER);
+
+        Good goodInFilteredList = model.getFilteredGoodList().get(INDEX_FIRST_SUPPLIER.getZeroBased());
+        Good editedGood = new GoodBuilder(goodInFilteredList).withThreshold(100).build();
+        SetThresholdCommand setThresholdCommand = new SetThresholdCommand(INDEX_FIRST_SUPPLIER,
+                editedGood.getThreshold());
+
+        String expectedMessage = String.format(SetThresholdCommand.MESSAGE_SUCCESS,
+                100, editedGood.getGoodName().fullGoodName);
+
+        Model expectedModel = new ModelManager(getTypicalAddressBook(), new Inventory(model.getInventory()),
+                getTypicalTransactionHistory(), new UserPrefs());
+        expectedModel.setGood(model.getFilteredGoodList().get(0), editedGood);
+
+        assertCommandSuccess(setThresholdCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidGoodIndexUnfilteredList_failure() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredGoodList().size() + 1);
+        SetThresholdCommand setThresholdCommand = new SetThresholdCommand(outOfBoundIndex, VALID_THRESHOLD);
+        assertCommandFailure(setThresholdCommand, model, Messages.MESSAGE_INVALID_GOOD_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Edit filtered list where index is larger than size of filtered list,
+     * but smaller than size of address book
+     */
+    @Test
+    public void execute_invalidGoodIndexFilteredList_failure() {
+        showGoodAtIndex(model, INDEX_FIRST_SUPPLIER);
+        Index outOfBoundIndex = INDEX_SECOND_SUPPLIER;
+        // ensures that outOfBoundIndex is still in bounds of inventory list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getInventory().getReadOnlyList().size());
+
+        SetThresholdCommand setThresholdCommand = new SetThresholdCommand(outOfBoundIndex, VALID_THRESHOLD);
+
+        assertCommandFailure(setThresholdCommand, model, Messages.MESSAGE_INVALID_GOOD_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void equals() {
+        final SetThresholdCommand standardCommand = new SetThresholdCommand(INDEX_FIRST_SUPPLIER, VALID_THRESHOLD);
+
+        // same values -> returns true
+        GoodQuantity threshold = new GoodQuantity(VALID_THRESHOLD.goodQuantity);
+        SetThresholdCommand commandWithSameValues = new SetThresholdCommand(INDEX_FIRST_SUPPLIER, threshold);
```

Comment:
Is there a need to define this new threshold?

Reply:
ok. i wll fix that